apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -101,17 +132,24 @@ func (api *CredentialValidatorAPI) WatchCredential(ctx context.Context, tag para
 	if err != nil {
 		return fail(err)
 	}
-	// Is credential used by the model that has created this backend?
-	modelCredentialTag, exists, err := api.backend.CloudCredentialTag()
+
+	modelInfo, err := api.modelInfoService.GetModelInfo(ctx)
 	if err != nil {
 		return fail(err)
 	}
-	if !exists || credentialTag != modelCredentialTag {
```

Comment:
The original code returned a permission error if the model has no credential. We need to do the same here.

Reply:
Sure thing Ian, just to verify is it sufficient if we check the the model credential name is empty to determine if the model has credential?

Reply:
> Sure thing Ian, just to verify is it sufficient if we check the the model credential name is empty to determine if the model has credential?

That will do for now. Ideally we'd have a better API